### PR TITLE
avoid ResourceWarning by using with-statement when calling open() #670

### PR DESCRIPTION
**Pull request by stroeder**

### CODE REVIEW

File: web/template.py [hunk 1]
```diff
@@ -1093,8 +1093,10 @@ def _load_template(self, name):
                 path, cache=self._cache is not None, base=self._base, **self._keywords
             )
         elif kind == "file":
+            with open(path, encoding="utf-8") as tmpl_file:
+                tmpl_str = tmpl_file.read()
             return Template(
-                open(path, encoding="utf-8").read(), filename=path, **self._keywords
+                tmpl_str, filename=path, **self._keywords
```
Comment (cclauss) [inline web/template.py:1099]:
```suggestion
            with open(path, encoding="utf-8") as tmpl_file:
                return Template(tmpl_file.read(), filename=path, **self._keywords)
```

Comment (stroeder) [reply web/template.py:1099]: Do you really want to place the return-statement in the body of the with-statement?

Comment (cclauss) [reply web/template.py:1099]: Yes.  This is no problem.  The context manager will close properly even with a return statement.

Comment (stroeder) [reply web/template.py]: Ok. Applied your suggestion.

File: web/template.py [hunk 2]
```diff
@@ -1093,9 +1093,10 @@ def _load_template(self, name):
                 path, cache=self._cache is not None, base=self._base, **self._keywords
             )
         elif kind == "file":
-            return Template(
-                open(path, encoding="utf-8").read(), filename=path, **self._keywords
-            )
+            with open(path, encoding="utf-8") as tmpl_file:
+                return Template(
+                    tmpl_file.read(), filename=path, **self._keywords
+                )
```
Comment (cclauss) [inline web/template.py:1099]:
```suggestion
                return Template(tmpl_file.read(), filename=path, **self._keywords)
```
To placate that pesky psf/black formatter.

**Comment by iredmail** [comment]
@cclauss Feel free to merge this PR if it's all good.

**Comment by iredmail** [comment]
@cclauss don't forgot to update ChangeLog file. :)

**Comment by cclauss** [comment]
The contributor should update the changelog, not the merger.

**Comment by iredmail** [comment]
We cannot wait for contributor to do everything, so if something is missing and we have some spare time, we'd better do it ourselves, this will help move on a lot faster.

**Comment by cclauss** [comment]
Also, this change is not worthy of a changelog entry as it has no impact on end users.

**Comment by iredmail** [comment]
It fixes a bug, so it's worth a line in ChangeLog.txt. :)

**Pull request merged**